feature/interface/aggregate/otg_tests/aggregate_all_not_viable_test/aggregate_all_not_forwarding_viable_test.go

```diff
@@ -0,0 +1,1025 @@
+		}
+		if err := checkBidirectionalTraffic(t, dut, dutPortList[(agg2.ateLagCount+1):]); err != nil {
+			t.Fatal(err)
+		}
+		if err := confirmNonViableForwardingTraffic(t, dut, ate, atePortList[2:(agg2.ateLagCount+1)], dutPortList[2:(agg2.ateLagCount+1)]); err != nil {
+			t.Fatal(err)
+		}
+		// Ensure Load Balancing 50:50 on LAG_2 and LAG_3
+		weights := trafficRXWeights(t, ate, []string{agg2.ateAggName, agg3.ateAggName}, flows[0])
+		for idx, weight := range equalDistributionWeights {
+			if got, want := weights[idx], weight; got < want-ecmpTolerance || got > want+ecmpTolerance {
+				t.Errorf("ECMP Percentage for Aggregate Index: %d: got %d, want %d", idx+1, got, want)
+			}
+		}
+		if ok := verifyTrafficFlow(t, ate, flows); !ok {
+			t.Fatal("Packet Dropped, LossPct for flow ")
+		}
+	})
+
+	t.Run("RT-5.7.2.2: Setting Forwarding-Viable to False for Lag2 all ports", func(t *testing.T) {
+		// Ensure ISIS Adjacency is up on LAG_2
+		if ok := awaitAdjacency(t, dut, aggIDs[1], oc.Isis_IsisInterfaceAdjState_UP); !ok {
+			t.Fatal("ISIS Adjacency is Down on LAG_2")
+		}
+		configForwardingViable(t, dut, dutPortList[1:2], false)
+		// Ensure ISIS Adjacency is Down on LAG_2
+		if ok := awaitAdjacency(t, dut, aggIDs[1], oc.Isis_IsisInterfaceAdjState_DOWN); !ok {
+			t.Fatal("ISIS Adjacency is Established on LAG_2")
+		}
+		otgutils.WaitForARP(t, ate.OTG(), top, "IPv4")
+		startTraffic(t, dut, ate, top)
+		if err := confirmNonViableForwardingTraffic(t, dut, ate, atePortList[1:(agg2.ateLagCount+1)], dutPortList[1:(agg2.ateLagCount+1)]); err != nil {
+			t.Fatal(err)
+		}
+		// Ensure that traffic from ATE port1 to pfx4 are discarded on DUT
+		if ok := verifyTrafficFlow(t, ate, flows[1:2]); !ok {
+			t.Fatal("Packet Dropped, LossPct for flow ")
+		}
+		// Ensure there is traffic received on DUT LAG_3
+		if got := validateLag3Traffic(t, dut, ate, dutPortList[(agg2.ateLagCount+1):]); got == false {
+			t.Fatal("Packets are not Received on LAG_3")
+		}
+		if ok := verifyTrafficFlow(t, ate, flows[0:1]); !ok {
+			t.Fatal("Packet Dropped, LossPct for flow ")
+		}
+	})
+
+	t.Run("RT-5.7.2.3: Setting Forwarding-Viable to True for Lag2 one of the port", func(t *testing.T) {
+		configForwardingViable(t, dut, dutPortList[agg2.ateLagCount:(agg2.ateLagCount+1)], true)
+		startTraffic(t, dut, ate, top)
+		if err := checkBidirectionalTraffic(t, dut, dutPortList[agg2.ateLagCount:]); err != nil {
+			t.Fatal(err)
+		}
+		if err := confirmNonViableForwardingTraffic(t, dut, ate, atePortList[1:agg2.ateLagCount], dutPortList[1:agg2.ateLagCount]); err != nil {
+			t.Fatal(err)
+		}
+		if ok := verifyTrafficFlow(t, ate, flows); !ok {
+			t.Fatal("Packet Dropped, LossPct for flow ")
+		}
+	})
+}
+
+// configureDUT configures DUT
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) []string {
+
+	t.Helper()
+	fptest.ConfigureDefaultNetworkInstance(t, dut)
+	if len(dut.Ports()) < 4 {
+		t.Fatalf("Testbed requires at least 4 ports, got %d", len(dut.Ports()))
```

Comment:
it needs to be len(dut.Ports()) <= 4 to cover   and also  change the error message to at least 5 ports

Reply:
@bkreddy143 
1. Default it use 4 port (1 LAG1 + 2 LAG2 + 1 LAG3) to work if testbed have 4 ports.
2. if the testbed have number of ports is < 4  it will throw error and come out from running.
3. If the testbed have number of ports > 4 it will bind the ports based on testbed.

Reply:
As this test will work for 4 ports as well i am marking it as resolved.